### PR TITLE
fix: Replaced Soft and Hard Deprecation Naming

### PR DESCRIPTION
**Pull request by josh-bagwell**
## Summary

Replacing the uses of `soft-deprecation` and `hard-deprecation` with `deprecation` and `removal`.

Fixes: #2042  

## Release Category
Documentation

### Release Note

Changed definition of `soft-deprecation` and `hard-deprecation` to `deprecation/deprecate` and `removal/remove`. 

---

## For the Reviewer

- [ ] PR title is short and descriptive
- [ ] PR summary describes the change (Fixes/Resolves linked correctly)

### CODE REVIEW

File: modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx [hunk 1]
```diff
@@ -76,29 +76,31 @@ help us maintain additional codemod transforms to make future upgrades easier.
...
-There are two types of deprecations: soft and hard.
+There is `deprecated` and `removal`.
 
-#### Soft Deprecation
+#### Deprecation
 
-A soft-deprecated component is still available with its full functionality, but it will have been
-renamed with a prefix to indicate its soft-deprecated status. It will also include a console warning
-to announce its deprecation. This warning will only be triggered on the component's initial render.
+We add the [@deprecated](https://jsdoc.app/tags-deprecated.html) from JSDoc to code that we plan to
```
Comment (willklein) [inline modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx:83]: Consider `@deprecated tag` for the link text.

File: modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx [hunk 2]
```diff
@@ -76,29 +76,31 @@ help us maintain additional codemod transforms to make future upgrades easier.
...
+deprecate in the near future. The component name will also have a strikethrough to indicate its
```
Comment (willklein) [inline modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx:84]: Where does it have a strikethrough? The context of this is missing and might not be safe to assume.

File: modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx [hunk 3]
```diff
@@ -146,14 +148,17 @@ export const CustomCookieBanner = (props: CustomCookieBannerProps) => {
 
 ### Header & Global Header
 
-We are [soft deprecating](#soft-deprecation) `Header` and `GlobalHeader` and their related exports.
-They has been renamed to `DeprecatedHeader` and `DeprecatedGlobalHeader`, respectively. You may
-continue to use these components exactly as you did in v5, but note that we plan to
-[hard-deprecate](#hard-deprecation) this package in Canvas Kit v7.
+We are [deprecating](#deprecation) `Header` and `GlobalHeader` and their related exports. They has
```
Comment (willklein) [inline modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx:151]: They *have been renamed

File: modules/docs/mdx/7.0-UPGRADE-GUIDE.mdx [hunk 4]
```diff
@@ -377,41 +377,39 @@ We've also changed the default depth for the following components:
...
-directly if you need additional help.
+Removals are deletions from our codebase and you can no longer consume this component. We've either
+promoted or replaced a component or utility. You will need to follow the method prescribed in our
+upgrade guide to update your application. Please reach out to our team directly if you need
+additional help.
 
 #### Cookie Banner
 
-We are [hard-deprecating](#hard-deprecation) `CookieBanner`. Please reference this
+We are [deprecating](#deprecation) `CookieBanner`. Please reference this
```
Comment (willklein) [inline modules/docs/mdx/7.0-UPGRADE-GUIDE.mdx:400]: I think where we have "hard-deprecating" it should be removing?

File: modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx (hunk 1 above)
Comment (josh-bagwell) [reply modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx:83]: Updated

File: modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx (hunk 2 above)
Comment (josh-bagwell) [reply modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx:84]: The strikethrough is the JSX Name of the component but I removed the mention of it since it's confusing.

File: modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx (hunk 3 above)
Comment (josh-bagwell) [reply modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx:151]: Updated

File: modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx [hunk 5]
```diff
@@ -146,14 +148,17 @@ export const CustomCookieBanner = (props: CustomCookieBannerProps) => {
...
+been renamed to `DeprecatedHeader` and `DeprecatedGlobalHeader`, respectively. You may continue to
+use these components exactly as you did in v5, but note that we plan to [remove](#removal) this
+package in Canvas Kit v7.
```
[the remark this answers is not in the document]
Comment (josh-bagwell) [reply modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx:154]: Done and updated

File: modules/docs/mdx/7.0-UPGRADE-GUIDE.mdx (hunk 4 above)
Comment (josh-bagwell) [reply modules/docs/mdx/7.0-UPGRADE-GUIDE.mdx:400]: Correct and has been updated

File: modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx [hunk 6]
```diff
@@ -76,29 +76,30 @@ help us maintain additional codemod transforms to make future upgrades easier.
...
-There are two types of deprecations: soft and hard.
+There is `deprecated` and `removal`.
 
-#### Soft Deprecation
+#### Deprecation
 
-A soft-deprecated component is still available with its full functionality, but it will have been
-renamed with a prefix to indicate its soft-deprecated status. It will also include a console warning
-to announce its deprecation. This warning will only be triggered on the component's initial render.
+We add the [@deprecated tag](https://jsdoc.app/tags-deprecated.html) from JSDoc to code that we plan
```
Comment (willklein) [inline modules/docs/mdx/6.0-UPGRADE-GUIDE.mdx:83]: Technically we don't do this yet, and it doesn't correctly describe what we did in v6 deprecations. I think we need content like: "In v6, we renamed components ..." with the text here following: "In v7, we no longer rename components, but we add the...."

File: modules/docs/mdx/7.0-UPGRADE-GUIDE.mdx [hunk 7]
```diff
@@ -377,41 +377,38 @@ We've also changed the default depth for the following components:
 
 ### Component Deprecations
 
-#### Deprecation Types
+### Deprecation Types
```
Comment (willklein) [inline modules/docs/mdx/7.0-UPGRADE-GUIDE.mdx:380]: This should still be #### i.e. an h4

File: modules/docs/mdx/7.0-UPGRADE-GUIDE.mdx [hunk 8]
```diff
@@ -377,41 +377,38 @@ We've also changed the default depth for the following components:
...
+Removals are deletions from our codebase and you can no longer consume this component. We've either
+promoted or replaced a component or utility. You will need to follow the method prescribed in our
+upgrade guide to update your application. Please reach out to our team directly if you need
+additional help.
 
 #### Cookie Banner
 
-We are [hard-deprecating](#hard-deprecation) `CookieBanner`. Please reference this
+We are [removing](#removal) `CookieBanner`. Please reference this
 [example](https://workday.github.io/canvas-kit/?path=/story/examples-cookiebanner-react--basic) to
```
Comment (willklein) [inline modules/docs/mdx/7.0-UPGRADE-GUIDE.mdx:400]: I know this isn't quite the focus of this PR, but these story links don't work anymore.

File: modules/docs/mdx/7.0-UPGRADE-GUIDE.mdx [hunk 9]
```diff
@@ -377,41 +377,38 @@ We've also changed the default depth for the following components:
...
 migrate away from `CookieBanner`.
 
 #### Page Header
 
-We are [hard-deprecating](#hard-deprecation) `PageHeader`. Please reference this
+We are [removing](#removal) `PageHeader`. Please reference this
 [example](https://workday.github.io/canvas-kit/?path=/story/examples-pageheader-react--basic) to
```
Comment (willklein) [inline modules/docs/mdx/7.0-UPGRADE-GUIDE.mdx:406]: Story link doesn't work.

File: modules/docs/mdx/7.0-UPGRADE-GUIDE.mdx [hunk 10]
```diff
@@ -377,41 +377,38 @@ We've also changed the default depth for the following components:
...
 migrate away from `PageHeader`.
 
 #### Header / Global Header
 
-We are [hard-deprecating](#hard-deprecation) `Header`. Please reference this
+We are [removing](#removal) `Header`. Please reference this
 [example](https://workday.github.io/canvas-kit/?path=/story/examples-globalheader-react--basic) to
```
Comment (willklein) [inline modules/docs/mdx/7.0-UPGRADE-GUIDE.mdx:412]: Story link doesn't work.

**Comment by josh-bagwell** [comment]
Removed all changes from previous commit and added new items to v9 Upgrade Guide

File: modules/codemod/lib/v8/spec/softDeprecateDrawer.spec.ts [hunk 11]
```diff
@@ -153,15 +153,15 @@ describe('Canvas Kit Deprecate Drawer Codemod', () => {
...
+
       type CustomDrawerHeaderProps = DrawerHeaderProps;
       interface AnotherDrawerHeaderProps extends DrawerHeaderProps {
         specialProp?: boolean;
       }
       `;
       const expected = `
       import { DeprecatedDrawerHeaderProps } from '@workday/canvas-kit-labs-react/drawer';
-  
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecateDrawer.spec.ts:164]: You can probably check out this file since this is prettiers work

File: modules/codemod/lib/v8/spec/softDeprecateLayout.spec.ts [hunk 12]
```diff
@@ -149,15 +149,15 @@ describe('Canvas Kit Deprecate Layout Codemod', () => {
...
+
       type CustomColumnProps = ColumnProps;
       interface AnotherColumnProps extends ColumnProps {
         specialProp?: boolean;
       }
       `;
       const expected = `
       import { DeprecatedColumnProps } from '@workday/canvas-kit-react/layout';
-  
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecateLayout.spec.ts:160]: same here, just undo these changes

File: modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts [hunk 13]
```diff
@@ -110,15 +110,15 @@ describe('Canvas Kit Deprecate Preview Menu Codemod', () => {
...
+
         type CustomMenuState = MenuState;
         interface AnotherMenuState extends MenuState {
           specialProp?: boolean;
         }
       `;
       const expected = `
         import { DeprecatedMenuState } from '@workday/canvas-kit-preview-react/menu';
-    
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts:121]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts [hunk 14]
```diff
@@ -110,15 +110,15 @@ describe('Canvas Kit Deprecate Preview Menu Codemod', () => {
     it('should properly transform type reference identifiers', () => {
       const input = `
         import { MenuState } from '@workday/canvas-kit-preview-react/menu';
-    
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts:113]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts [hunk 15]
```diff
@@ -89,15 +89,15 @@ describe('Canvas Kit Deprecate Preview Menu Codemod', () => {
...
         type CustomMenuItemProps = MenuItemProps;
         interface AnotherMenuItemProps extends MenuItemProps {
           specialProp?: boolean;
         }
       `;
       const expected = `
         import { DeprecatedMenuItemProps } from '@workday/canvas-kit-preview-react/menu';
-    
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts:100]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts [hunk 16]
```diff
@@ -89,15 +89,15 @@ describe('Canvas Kit Deprecate Preview Menu Codemod', () => {
     it('should properly transform type reference identifiers', () => {
       const input = `
         import { MenuItemProps } from '@workday/canvas-kit-preview-react/menu';
-    
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts:92]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts [hunk 17]
```diff
@@ -68,15 +68,15 @@ describe('Canvas Kit Deprecate Preview Menu Codemod', () => {
...
         type CustomMenuProps = MenuProps;
         interface AnotherMenuProps extends MenuProps {
           specialProp?: boolean;
         }
       `;
       const expected = `
         import { DeprecatedMenuProps } from '@workday/canvas-kit-preview-react/menu';
-    
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts:79]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts [hunk 18]
```diff
@@ -68,15 +68,15 @@ describe('Canvas Kit Deprecate Preview Menu Codemod', () => {
     it('should properly transform type reference identifiers', () => {
       const input = `
         import { MenuProps } from '@workday/canvas-kit-preview-react/menu';
-    
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts:71]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts [hunk 19]
```diff
@@ -48,7 +48,7 @@ describe('Canvas Kit Deprecate Preview Menu Codemod', () => {
 
       const expected = `
         import {DeprecatedMenu, DeprecatedMenuItem} from '@workday/canvas-kit-preview-react/menu';
-    
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts:51]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts [hunk 20]
```diff
@@ -32,7 +32,7 @@ describe('Canvas Kit Deprecate Preview Menu Codemod', () => {
     it('should properly transform Menu JSX identifiers', () => {
       const input = `
         import {Menu, MenuItem} from '@workday/canvas-kit-preview-react/menu';
-    
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecatePreviewMenu.spec.ts:35]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecateLayout.spec.ts (hunk 12 above)
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecateLayout.spec.ts:160]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecateLayout.spec.ts [hunk 21]
```diff
@@ -149,15 +149,15 @@ describe('Canvas Kit Deprecate Layout Codemod', () => {
     it('should properly transform type reference identifiers', () => {
       const input = `
       import { ColumnProps } from '@workday/canvas-kit-react/layout';
-  
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecateLayout.spec.ts:152]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecateLayout.spec.ts [hunk 22]
```diff
@@ -128,15 +128,15 @@ describe('Canvas Kit Deprecate Layout Codemod', () => {
...
       type CustomLayoutProps = LayoutProps;
       interface AnotherLayoutProps extends LayoutProps {
         specialProp?: boolean;
       }
       `;
       const expected = `
       import { DeprecatedLayoutProps } from '@workday/canvas-kit-react/layout';
-  
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecateLayout.spec.ts:139]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecateLayout.spec.ts [hunk 23]
```diff
@@ -128,15 +128,15 @@ describe('Canvas Kit Deprecate Layout Codemod', () => {
     it('should properly transform type reference identifiers', () => {
       const input = `
       import { LayoutProps } from '@workday/canvas-kit-react/layout';
-  
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecateLayout.spec.ts:131]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecateLayout.spec.ts [hunk 24]
```diff
@@ -86,7 +86,7 @@ describe('Canvas Kit Deprecate Layout Codemod', () => {
 
       const expected = `
       import {DeprecatedLayout} from '@workday/canvas-kit-react/layout';
-  
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecateLayout.spec.ts:89]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecateLayout.spec.ts [hunk 25]
```diff
@@ -74,7 +74,7 @@ describe('Canvas Kit Deprecate Layout Codemod', () => {
     it('should properly transform Layout JSX identifiers', () => {
       const input = `
       import {Layout} from '@workday/canvas-kit-react/layout';
-  
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecateLayout.spec.ts:77]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecateDrawer.spec.ts (hunk 11 above)
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecateDrawer.spec.ts:164]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecateDrawer.spec.ts [hunk 26]
```diff
@@ -153,15 +153,15 @@ describe('Canvas Kit Deprecate Drawer Codemod', () => {
     it('should properly transform type reference identifiers', () => {
       const input = `
       import { DrawerHeaderProps } from '@workday/canvas-kit-labs-react/drawer';
-  
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecateDrawer.spec.ts:156]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecateDrawer.spec.ts [hunk 27]
```diff
@@ -132,15 +132,15 @@ describe('Canvas Kit Deprecate Drawer Codemod', () => {
...
       type CustomDrawerProps = DrawerProps;
       interface AnotherDrawerProps extends DrawerProps {
         specialProp?: boolean;
       }
       `;
       const expected = `
       import { DeprecatedDrawerProps } from '@workday/canvas-kit-labs-react/drawer';
-  
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecateDrawer.spec.ts:143]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecateDrawer.spec.ts [hunk 28]
```diff
@@ -132,15 +132,15 @@ describe('Canvas Kit Deprecate Drawer Codemod', () => {
     it('should properly transform type reference identifiers', () => {
       const input = `
       import { DrawerProps } from '@workday/canvas-kit-labs-react/drawer';
-  
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecateDrawer.spec.ts:135]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecateDrawer.spec.ts [hunk 29]
```diff
@@ -80,7 +80,7 @@ describe('Canvas Kit Deprecate Drawer Codemod', () => {
 
       const expected = `
       import {DeprecatedDrawer, DeprecatedDrawerDirection} from '@workday/canvas-kit-labs-react/drawer';
-  
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecateDrawer.spec.ts:83]:
```suggestion
```

File: modules/codemod/lib/v8/spec/softDeprecateDrawer.spec.ts [hunk 30]
```diff
@@ -64,7 +64,7 @@ describe('Canvas Kit Deprecate Drawer Codemod', () => {
     it('should properly transform Drawer JSX identifiers', () => {
       const input = `
       import {Drawer, DrawerDirection} from '@workday/canvas-kit-labs-react/drawer';
-  
+
```
Comment (mannycarrera4) [inline modules/codemod/lib/v8/spec/softDeprecateDrawer.spec.ts:67]:
```suggestion
```
